atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/timestamp/TimestampServiceTests.java

```diff
@@ -0,0 +1,141 @@
+ */
+package com.palantir.atlasdb.timestamp;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.assertj.core.api.Assertions;
+
+import com.palantir.timestamp.TimestampMigrationService;
+import com.palantir.timestamp.TimestampRange;
+import com.palantir.timestamp.TimestampService;
+
+public class TimestampServiceTests {
+    private static final long ONE_MILLION = 1000 * 1000;
+    private static final long TWO_MILLION = 2 * ONE_MILLION;
+
+    private static ExecutorService executor = Executors.newFixedThreadPool(16);
+
+    private TimestampServiceTests() {
+
+    }
+
+    public static void timestampsAreReturnedInOrder(TimestampService timestampService) {
+        List<Long> timestamps = new ArrayList<>();
+
+        timestamps.add(timestampService.getFreshTimestamp());
+        timestamps.add(timestampService.getFreshTimestamp());
+
+        assertThat(timestamps.get(0), lessThan(timestamps.get(1)));
+    }
+
+    public static void canRequestTimestampRangeWithGetFreshTimestamps(TimestampService timestampService) {
+        int expectedNumTimestamps = 5;
+        TimestampRange range = timestampService.getFreshTimestamps(expectedNumTimestamps);
+
+        long actualNumTimestamps = range.getUpperBound() - range.getLowerBound() + 1;
+        assertThat(
+                String.format("Expected %d timestamps, got %d timestamps. (The returned range was: %d-%d)",
+                        expectedNumTimestamps, actualNumTimestamps, range.getLowerBound(), range.getUpperBound()),
+                (int) actualNumTimestamps,
+                is(both(greaterThanOrEqualTo(1)).and(lessThanOrEqualTo(expectedNumTimestamps))));
+    }
+
+    public static void timestampRangesAreReturnedInNonOverlappingOrder(TimestampService timestampService) {
+        List<TimestampRange> timestampRanges = new ArrayList<>();
+
+        timestampRanges.add(timestampService.getFreshTimestamps(10));
+        timestampRanges.add(timestampService.getFreshTimestamps(10));
+
+        long firstUpperBound = timestampRanges.get(0).getUpperBound();
+        long secondLowerBound = timestampRanges.get(1).getLowerBound();
+
+        assertThat(firstUpperBound, is(lessThan(secondLowerBound)));
+    }
+
+    public static void canRequestMoreTimestampsThanAreAllocatedAtOnce(TimestampService timestampService) {
+        for (int i = 0; i < ONE_MILLION / 1000; i++) {
```

Comment:
why not just use `1000`, or else define `ONE_THOUSAND` with the other constants?